### PR TITLE
Core 485 throw an error upon startup, if smart agent account config is invalid

### PR DESCRIPTION
**Pull request by OmairLiaquatAli**
- [CORE-485]

[CORE-485]: https://evannetwork.atlassian.net/browse/CORE-485

### CODE REVIEW

File: lib/smart-agent.js [hunk 1]
```diff
@@ -39,6 +39,19 @@ module.exports = class SmartAgent {
     // config fore core related components
     this.coreConfig = api.config.smartAgentCore
 
+    console.dir(this.config)
```
Comment (wulfraem) [inline lib/smart-agent.js]:
```suggestion
```

File: test/startupChecks.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  limitations under the License.
+*/
+
+/* global describe, before, after, it */
+
+'use strict'
+
+const chai = require('chai')
+const chaiAsPromised = require('chai-as-promised');
+//const dirtyChai = require('dirty-chai')
```
Comment (wulfraem) [inline test/startupChecks.js]:
```suggestion
```

File: test/startupChecks.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+  it('can not create a new class without account configuration', async () => {
+    class TestClass extends api.smartAgents.SmartAgent {
+      constructor(config) {
+        super(config)
+      }
+    }
+
+    const test = new TestClass({});
+    console.log(test)
```
Comment (wulfraem) [inline test/startupChecks.js]:
```suggestion
```

File: test/startupChecks.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    class TestClass extends api.smartAgents.SmartAgent {
+      constructor(config) {
+        super(config)
+      }
+    }
+
+    const test = new TestClass({
+          ethAccount:'0x0000000000000000000000000000000000000000',
+        });
+    console.log(test)
```
Comment (wulfraem) [inline test/startupChecks.js]:
```suggestion
```

File: test/startupChecks.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    await expect(test.initialize()).to.be.rejectedWith('accountId does not exist') 
+  })
+
+  it('can not create a new class with out private key', async () => {
+    class TestClass extends api.smartAgents.SmartAgent {
+      constructor(config) {
+        super(config)
+      }
+    }
+    // add temporary account to api
```
Comment (wulfraem) [inline test/startupChecks.js]:
```suggestion
    // remove account from config (if present)
```

File: test/startupChecks.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    //api.config.ethAccounts['0x15B10D6521D17a205eaC67b41770c7F447431d89'] = ''
```
Comment (wulfraem) [inline test/startupChecks.js]:
```suggestion
```

File: test/startupChecks.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    delete api.config.ethAccounts['0x15B10D6521D17a205eaC67b41770c7F447431d89']
+    const test = new TestClass({
+      ethAccount: '0x15B10D6521D17a205eaC67b41770c7F447431d89',
+    });
+    console.log(test)
```
Comment (wulfraem) [inline test/startupChecks.js]:
```suggestion
```

File: test/startupChecks.js [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+      constructor(config) {
+        super(config)
+      }
+    }
+    // add temporary account to api
+    api.config.ethAccounts['0x15B10D6521D17a205eaC67b41770c7F447431d89'] = '185f24bfafc1493b005de1d350d6853eca0555f678013dd3f555cc939e7a1ac2'
+    const test = new TestClass({
+      ethAccount: '0x15B10D6521D17a205eaC67b41770c7F447431d89',
+    });
+    console.log(test)
```
Comment (wulfraem) [inline test/startupChecks.js]:
```suggestion
```

File: test/startupChecks.js [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+
+  it('can not create a new class with accountId 0x000', async () => {
+    class TestClass extends api.smartAgents.SmartAgent {
+      constructor(config) {
+        super(config)
+      }
+    }
+
+    const test = new TestClass({
+          ethAccount:'0x0000000000000000000000000000000000000000',
```
Comment (wulfraem) [inline test/startupChecks.js]:
```suggestion
          ethAccount:'0x0000000000000000000000000000000000000000'
```

File: test/startupChecks.js [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+        });
```
Comment (wulfraem) [inline test/startupChecks.js]:
```suggestion
        })
```

File: test/startupChecks.js [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+    class TestClass extends api.smartAgents.SmartAgent {
+      constructor(config) {
+        super(config)
+      }
+    }
+    // add temporary account to api
+    //api.config.ethAccounts['0x15B10D6521D17a205eaC67b41770c7F447431d89'] = ''
+    delete api.config.ethAccounts['0x15B10D6521D17a205eaC67b41770c7F447431d89']
+    const test = new TestClass({
+      ethAccount: '0x15B10D6521D17a205eaC67b41770c7F447431d89',
```
Comment (wulfraem) [inline test/startupChecks.js]:
```suggestion
      ethAccount: '0x15B10D6521D17a205eaC67b41770c7F447431d89'
```

File: test/startupChecks.js [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+    });
```
Comment (wulfraem) [inline test/startupChecks.js]:
```suggestion
    })
```

File: test/startupChecks.js [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+  it('can create a new class with proper configurations', async () => {
+    class TestClass extends api.smartAgents.SmartAgent {
+      constructor(config) {
+        super(config)
+      }
+    }
+    // add temporary account to api
+    api.config.ethAccounts['0x15B10D6521D17a205eaC67b41770c7F447431d89'] = '185f24bfafc1493b005de1d350d6853eca0555f678013dd3f555cc939e7a1ac2'
+    const test = new TestClass({
+      ethAccount: '0x15B10D6521D17a205eaC67b41770c7F447431d89',
```
Comment (wulfraem) [inline test/startupChecks.js]:
```suggestion
      ethAccount: '0x15B10D6521D17a205eaC67b41770c7F447431d89'
```

File: test/startupChecks.js [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+    });
```
Comment (wulfraem) [inline test/startupChecks.js]:
```suggestion
    })
```

**Review by wulfraem — CHANGES_REQUESTED** [review]

File: lib/smart-agent.js [hunk 15]
```diff
@@ -39,6 +39,19 @@ module.exports = class SmartAgent {
...
+    if(!this.config.ethAccount){
```
Comment (Tschuck) [inline lib/smart-agent.js]:
```suggestion
    if (!this.config.ethAccount) {
```

File: lib/smart-agent.js [hunk 16]
```diff
@@ -39,6 +39,19 @@ module.exports = class SmartAgent {
...
+      throw new Error('No account has been configured')
+    }
+
+    if(this.config.ethAccount === '0x0000000000000000000000000000000000000000'){
```
Comment (Tschuck) [inline lib/smart-agent.js]:
```suggestion
    if (this.config.ethAccount === '0x0000000000000000000000000000000000000000') {
```

File: lib/smart-agent.js [hunk 17]
```diff
@@ -39,6 +39,19 @@ module.exports = class SmartAgent {
...
+      throw new Error('accountId does not exist')
+    }
+
+    if(!api.config.ethAccounts[this.config.ethAccount]){
```
Comment (Tschuck) [inline lib/smart-agent.js]:
```suggestion
    if (!api.config.ethAccounts[this.config.ethAccount]) {
```

**Review by Tschuck — CHANGES_REQUESTED** [review]

**Review by wulfraem — APPROVED** [review]

File: test/startupChecks.js [hunk 18]
```diff
@@ -0,0 +1,94 @@
...
+
+  it('can not create a new class with accountId 0x000', async () => {
+    class TestClass extends api.smartAgents.SmartAgent {
+      constructor(config) {
+        super(config)
+      }
+    }
+
+    const test = new TestClass({
+          ethAccount:'0x0000000000000000000000000000000000000000'
```
Comment (Tschuck) [inline test/startupChecks.js]:
```suggestion
      ethAccount:'0x0000000000000000000000000000000000000000'
```

File: test/startupChecks.js [hunk 19]
```diff
@@ -0,0 +1,94 @@
...
+        })
```
Comment (Tschuck) [inline test/startupChecks.js]:
```suggestion
    })
```

**Review by Tschuck — DISMISSED** [review]

File: test/startupChecks.js [hunk 20]
```diff
@@ -0,0 +1,94 @@
...
+      constructor(config) {
+        super(config)
+      }
+    }
+    // add temporary account to api
+    api.config.ethAccounts['0x15B10D6521D17a205eaC67b41770c7F447431d89'] = '185f24bfafc1493b005de1d350d6853eca0555f678013dd3f555cc939e7a1ac2'
+    const test = new TestClass({
+      ethAccount: '0x15B10D6521D17a205eaC67b41770c7F447431d89'
+    })
+    await expect(test.initialize()).to.not.be.rejected
```
Comment (karladler) [inline test/startupChecks.js]: this looks like more correct english, but I didn't tested it 

```suggestion
    await expect(test.initialize()).not.to.be.rejected 
```

**Review by S3bb1 — APPROVED** [review]

**Pull request merged**